### PR TITLE
[server] Sanitise image creation

### PR DESCRIPTION
**Pull request by pblottiere**
## Description
Include a few sentences describing the overall goals for this PR (pull request). If applicable also add screenshots.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by pblottiere** [comment]
Hi @elpaso,

What do you think about this PR?

File: src/server/services/wms/qgswmsgetfeatureinfo.cpp [hunk 1]
```diff
@@ -31,6 +31,15 @@ namespace QgsWms
     // get wms parameters from query
     QgsWmsParameters parameters( QUrlQuery( request.url() ) );
 
+    // WIDTH and HEIGHT are not mandatory, but we need to set a default size
```
Comment (elpaso) [inline src/server/services/wms/qgswmsgetfeatureinfo.cpp]: What if only one of the values is set?

**Review by elpaso — APPROVED** [review]
One thing I would take into account is that in case `HEIGHT` and `WIDTH` are not set but `FILTER_GEOM` is, it would probably be appropriate to recalculate the `HEIGHT` and `WIDTH` from a standard size (say 800 px on the horizontal axis or 600 px on the vertical one) instead of 10x10 and make a proportion  from the bounding box of `FILTER_GEOM`.

**Comment by pblottiere** [comment]
Hi @elpaso,

I think I took into account your comments. Can you take a look to ensure that I correctly understood your suggestions?

And thanks for the review!

File: src/server/services/wms/qgswmsgetfeatureinfo.cpp (hunk 1 above)
Comment (elpaso) [reply src/server/services/wms/qgswmsgetfeatureinfo.cpp]: LGTM, thanks!

**Pull request merged**